lints.rs

```diff
@@ -13,36 +13,45 @@
     clippy::exit,
     clippy::explicit_into_iter_loop,
     clippy::filter_map_next,
+    clippy::float_cmp_const,
     clippy::fn_params_excessive_bools,
+    clippy::get_unwrap,
     clippy::if_let_mutex,
     clippy::imprecise_flops,
     clippy::inefficient_to_string,
     clippy::let_unit_value,
     clippy::linkedlist,
     clippy::lossy_float_literal,
     clippy::macro_use_imports,
+    clippy::map_err_ignore,
     clippy::map_flatten,
     clippy::map_unwrap_or,
     clippy::match_on_vec_items,
     clippy::match_wildcard_for_single_variants,
     clippy::mem_forget,
     clippy::mismatched_target_os,
+    clippy::mutex_integer,
     clippy::needless_borrow,
     clippy::needless_continue,
     clippy::option_option,
+    clippy::print_stdout,
     clippy::pub_enum_variant_names,
     clippy::ref_option_ref,
     clippy::rest_pat_in_fully_bound_structs,
+    clippy::string_add,
     clippy::string_to_string,
     clippy::suboptimal_flops,
     clippy::todo,
+    clippy::trivially_copy_pass_by_ref,
```

Comment:
`trivially_copy_pass_by_ref ` can catch some good small optimizations here and there, but as it want switches from references to values that can be a bit intrusive in places so not fully sure it is a good idea to generally enforce.

Reply:
It has also found a `.clone()` on a simple `enum` because that wasn't marked `Copy` at first :grimacing: